api/v1beta1/aerospikecluster_validating_webhook_test.go

```diff
@@ -0,0 +1,187 @@
+package v1beta1
+
+import (
+	"testing"
+	"fmt"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var colorGreen = "\033[32m"
+var colorRed = "\033[31m"
+var colorReset = "\033[0m"
+
+// To test: go to /api/v1beta1 and run this command:
+// go test -run TestEnhancePesistentNamespaces
+func TestEnhancePesistentNamespaces(t *testing.T) {
+	TestAddPesistentNamespaceWithNotUsedDevices(t)
+	TestAddPesistentNamespaceWithAlreadyUsedDevices(t)
+	TestAddMultiplePesistentNamespacesWithNotUsedDevices(t)
+	TestAddMultiplePesistentNamespacesWithAlreadyUsedDevices(t)
+}
+
+func TestAddPesistentNamespaceWithNotUsedDevices(t *testing.T) {
+	oldNsConf := map[string][]string {
+		"namespace-0": []string{"/opt/data01/xvd01", "/opt/data01/xvd02"},
+	}
+	newNsConf := map[string][]string {
+		"namespace-0": []string{"/opt/data01/xvd01", "/opt/data01/xvd02"},
+		"new-namespace": []string{"/opt/data02/xvd01", "/opt/data02/xvd02"},
+	}
+
+	aslog := logf.Log.WithName("Test validateNsConfUpdate")
+	oldNamespaceConfig, newNamespaceConfig := prepareNsConfigurations(oldNsConf, newNsConf)
+
+	err := validateNsConfUpdate(aslog, &newNamespaceConfig, &oldNamespaceConfig)
+	if err != nil {
+		fmt.Println(string(colorRed), "TestAddPesistentNamespaceWithNotUsedDevices failed.", string(colorReset))
```

Comment:
I think you aren't supposed to log test failed / test passed. It is supposed to be done by test runner.

Reply:
Indeed, it's already done. I just wanted to add some colors (there is no color when the test fails or passes)

Reply:
Done